File: python/0071.py
class Solution:
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        a = path.split("/")
        now = []
        for i in a:
            if not i: continue
            if i == '.':
                continue
            elif i == '..':
                if now:
                    now.pop()
            else:
                now.append(i)
        return '/' + '/'.join(now)


if __name__ == '__main__':
    print(Solution().simplifyPath("/a/../b"))

File: rust/src/bin/0071.rs
struct Solution;

impl Solution {
    pub fn simplify_path(path: String) -> String {
        let ans: Vec<&str> = path.split("/").collect();
        let mut sta = vec![];
        for i in ans {
            if i.len() == 0 {
                continue;
            }
            if i == ".".to_string() {
                continue;
            }
            if i == "..".to_string() {
                sta.pop();
                continue;
            }
            sta.push(i);
        }
        let mut a: String = sta.join("/");
        a.insert(0, '/');
        a
    }
}

fn main() {
    let ans = Solution::simplify_path("/haha/../one/./three".to_string());
    println!("{}", ans);
}